api/namex/models/request.py

```diff
@@ -111,6 +111,19 @@ class Request(db.Model):
     REQUEST_FURNISHED = 'Y'
 
     # properties
+    @property
+    def has_consumed_name(self):
```

Comment:
hmm. wondering why this is not part of the names model instead?

Reply:
Convenience eg.
`
def display_cancel_action(nr_model=None):
    try:
        if (nr_model and nr_model.stateCd == State.CANCELLED) or (nr_model.stateCd not in State.CANCELLABLE_STATES):
            return False

        if nr_model and (nr_model.is_expired or nr_model.has_consumed_name):
            return False

        return True
    except Exception as err:
        raise NameRequestActionError(err)
`